src/nvim/main.c

```diff
@@ -250,6 +250,11 @@ int main(int argc, char **argv)
 
   argv0 = argv[0];
 
+  if (!appname_is_valid()) {
+    os_errmsg("Invalid $NVIM_APPNAME (see `:help $NVIM_APPNAME`)\n");
```

Comment:
"foo (see :help foo)" is redundant

we probably want a EXXX error code for this if we're going to mention it in the docs.

```suggestion
    os_errmsg("$NVIM_APPNAME must not contain path separators\n");
```

Reply:
I removed references to the documentation as suggested and attempted to make the requirements both broader and clearer (to cover cases where users might attempt to use characters forbidden by their filesystem, like `:` under NTFS).